### PR TITLE
Updates ``~sunpy.map.mapbase.draw_grid()`` to allow disabling the axes labels and the ticks on the top and right axes.

### LINKED ISSUE #3208: Add an annotate=False kwarg to map.draw_grid
This would disable the axes labels and the ticks on the top and right axes.

For example: http://docs.sunpy.org/en/latest/generated/gallery/map/submaps_and_cropping.html#sphx-glr-generated-gallery-map-submaps-and-cropping-py

In the submap, you have the top and right axes which correspond to the grid and we would like to turn that off.

### PR DESCRIPTION
**Pull request by Raahul-Singh**
### Description

Fixes #3208

### CODE REVIEW

**Comment by Raahul-Singh** [comment]
@nabobalis I did not request for reviews. They were added automatically. I do not understand this.

**Comment by nabobalis** [comment]
> @nabobalis I did not request for reviews. They were added automatically. I do not understand this.

Its how its setup now from outside, any new PR will have it now. It just was not working recently.

**Comment by Raahul-Singh** [comment]
> > @nabobalis I did not request for reviews. They were added automatically. I do not understand this.
> 
> Its how its setup now from our side, any new PR will have it now. It just was not working recently.

Oh, cool. That makes sense. :sweat_smile:

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -1450,7 +1450,7 @@ def cmap(self):
         return cmap
 
     @u.quantity_input
-    def draw_grid(self, axes=None, grid_spacing: u.deg = 15*u.deg, **kwargs):
+    def draw_grid(self, axes=None, grid_spacing: u.deg = 15*u.deg, annotate=False, **kwargs):
```
Comment (Cadair) [inline sunpy/map/mapbase.py]:
```suggestion
    def draw_grid(self, axes=None, grid_spacing: u.deg = 15*u.deg, annotate=True, **kwargs):
```

File: sunpy/visualization/wcsaxes_compat.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def default_wcs_grid(axes):
 
 
 @u.quantity_input
-def wcsaxes_heliographic_overlay(axes, grid_spacing: u.deg = 10*u.deg, **kwargs):
+def wcsaxes_heliographic_overlay(axes, grid_spacing: u.deg = 10*u.deg, annotate=False, **kwargs):
```
Comment (Cadair) [inline sunpy/visualization/wcsaxes_compat.py]:
```suggestion
def wcsaxes_heliographic_overlay(axes, grid_spacing: u.deg = 10*u.deg, annotate=True, **kwargs):
```

File: changelog/3673.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Updates ``~sunpy.map.mapbase.draw_grid()`` to allow disabling the axes labels and the ticks on the top and right axes.
```
Comment (Cadair) [inline changelog/3673.feature.rst]:
```suggestion
Updates `sunpy.map.GenericMap.draw_grid` to allow disabling the axes labels and the ticks on the top and right axes.
```

**Comment by Raahul-Singh** [comment]
> Could you post the figures that the new figure test generates? (just to make sure it's doing the right thing!)

Following the [original link](http://docs.sunpy.org/en/latest/generated/gallery/map/submaps_and_cropping.html#sphx-glr-generated-gallery-map-submaps-and-cropping-py) given in the issue,
`swap_map = sunpy.map.Map(sunpy.data.sample.SWAP_LEVEL1_IMAGE)`
`top_right = SkyCoord(0 * u.arcsec, -200 * u.arcsec, frame=swap_map.coordinate_frame)
 bottom_left = SkyCoord(-900 * u.arcsec, -900 * u.arcsec, frame=swap_map.coordinate_frame)
 swap_submap = swap_map.submap(bottom_left, top_right)`
`fig = plt.figure()`
`ax = fig.add_subplot(111, projection=swap_submap)`
`swap_submap.plot()`
`swap_submap.draw_limb()`
`swap_submap.draw_grid()`
`plt.show()`

will get,
before

modifying the above code,
`swap_submap.draw_grid(annotate=False)`

will give,
after

**Comment by nabobalis** [comment]
Need to upload the figure to the images repo. 

Question is, do we want this set to false by default?

**Comment by Cadair** [comment]
> Question is, do we want this set to false by default?

I vote no.

**Comment by nabobalis** [comment]
But look how ugly it is out of the box.

**Comment by Cadair** [comment]
We should come up with a better way to move the title, but other than that it's ok

**Pull request merged**